crud/src/main/java/com/redhat/lightblue/assoc/Conjunct.java

```diff
@@ -54,6 +65,202 @@
+            ItrState state=new ItrState(ConjunctType.value);
+            state.numEntities=0;
+            return state;
+        }
+
+        private ItrState oneField(QueryExpression q) {
+            ItrState state=new ItrState(ConjunctType.value);
+            state.e1=findOne(q).getFieldEntity();
+            state.numEntities=1;
+            return state;
+        }
+
+        private ItrState twoFields(QueryExpression q) {
+            ItrState state=new ItrState(ConjunctType.value);
+            QueryFieldInfo[] qfi=findTwo(q);
+            state.e1=qfi[0].getFieldEntity();
+            state.numEntities=1;
+            if(qfi[1]!=null&&qfi[1].getFieldEntity()!=state.e1) {
+                state.e2=qfi[1].getFieldEntity();
+                state.numEntities=2;
+                state.type=ConjunctType.relation;
+            } 
+            return state;
+        }
+
+        @Override
+        protected ItrState itrValueComparisonExpression(ValueComparisonExpression q, Path context) {
+            return oneField(q);
+        }
+
+        @Override
+        protected ItrState itrFieldComparisonExpression(FieldComparisonExpression q, Path context) {
+            return twoFields(q);
+        }
+
+        @Override
+        protected ItrState  itrRegexMatchExpression(RegexMatchExpression q, Path context) {
+            return oneField(q);
+        }
+
+        @Override
+        protected ItrState itrNaryValueRelationalExpression(NaryValueRelationalExpression q, Path context) {
+            return oneField(q);
+        }
+
+        @Override
+        protected ItrState  itrNaryFieldRelationalExpression(NaryFieldRelationalExpression q, Path context) {
+            return twoFields(q);
+        }
+
+        @Override
+        protected ItrState  itrArrayContainsExpression(ArrayContainsExpression q, Path context) {
+            return oneField(q);
+        }
+
+        @Override
+        protected ItrState  itrUnaryLogicalExpression(UnaryLogicalExpression q, Path context) {
+            return iterate(q.getQuery(),context);
+        }
+        
+        @Override
+        protected ItrState itrArrayMatchExpression(ArrayMatchExpression q, Path context) {
+            return iterate(q.getElemMatch(),context);
+        }
+
+        @Override
+        protected ItrState itrNaryLogicalExpression(NaryLogicalExpression q, Path context) {
+            ItrState ret=null;
+            boolean first=true;
+            for(QueryExpression subq:q.getQueries()) {
+                if(first) {
+                    ret=iterate(subq,context);
+                    first=false;
+                } else if(ret.type!=ConjunctType.complex) {                    
+                    ItrState qstate=iterate(subq,context);
+                    // We compare if this expression is compatible with existing assessment
+                    switch(qstate.type) {
+                    case value:
```

Comment:
There is a lot of logic in these case statements. Perhaps private methods might help the readability?